### PR TITLE
Fixes #34688 - Handle missing executable more gracefully

### PR DESCRIPTION
**Pull request by adamruzicka**
in ProcessManager and friends

This should also fix https://projects.theforeman.org/issues/31888 without us having to do any further changes to sp-ansible

### CODE REVIEW

**Comment by adamruzicka** [comment]
Not sure why travis is complainig, let's try once more
[test smart_proxy_dynflow]

File: lib/smart_proxy_dynflow/process_manager.rb [hunk 1]
```diff
@@ -158,8 +158,10 @@ def close
       # @return [void]
       def finish
         close
-        _pid, status = Process.wait2(@pid)
-        @status = status.exitstatus
+        unless @status
```
Comment (ezr-ondrej) [inline lib/smart_proxy_dynflow/process_manager.rb:161]: unless `pid == -1` ? :)

**Review by ezr-ondrej — APPROVED** [review]
One nit, but looks definitelly lovely to have ! :+1:

File: lib/smart_proxy_dynflow/process_manager.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/smart_proxy_dynflow/process_manager.rb]: Right

**Comment by adamruzicka** [comment]
Come on, mr. J
[test smart_proxy_dynflow]

**Pull request merged**